src/main/java/Validation.java

```diff
@@ -0,0 +1,64 @@
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Validation handles the initial validation of the user command inputs.
+ */
+public class Validation {
+
+    /**
+     * Checks if the user input is within the list of accepted commands.
+     * @param command User's command input.
+     * @throws DukeException On invalid input.
+     */
+    public static void checkValidCommand(String command) throws DukeException {
+        String[] validCommands = new String[] {"todo", "deadline", "event", "list", "bye", "done", "delete", "find"};
+        List<String> commands = Arrays.asList(validCommands);
+
+        int index = command.indexOf(' ');
+        String first = "";
+        if (index > -1) {
+            first = command.substring(0, index);
+        } else {
+            first = command;
+        }
+
+        if (!commands.contains(first.toLowerCase())) {
+            throw new DukeException(":( OOPS! I'm sorry, but I don't know what that means :-(");
+        } else {
+            String[] secondValidation = new String[] {"todo", "deadline", "event", "done", "delete", "find"};
+            List<String> secondListOfCommands = Arrays.asList(secondValidation);
+
+            if (secondListOfCommands.contains(first.toLowerCase())) {
+                if (index <= -1 || command.substring(index).isBlank()) {
```

Comment:
Great error checking here, but instead of nesting the if statements, perhaps they could be combined into one?